Upload z/OSMF workflow sample files

zhaocaih opened the pull request:
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Workflow Samples

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

Conner-IBM commented on zOSMF-Samples/workflow_sample_automation.xml:
```diff
@@ -0,0 +1,189 @@
+	<step name="subStep2">
+		<title>Second-level step 2</title>
+		<description>This is a second-level step 2.</description>
+		<!-- This step take subStep1_1 as its prerequisite step. -->
```
Grammar here seems odd.
```suggestion
		<!-- This prereqStep element enforces that subStep1_1 must be completed before this step can be run. -->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_automation.xml:
```diff
@@ -0,0 +1,189 @@
+id $instance-st_user
+			</inlineTemplate>
+       		<submitAs>shell-JCL</submitAs>
+       		<!-- When user save unix file more than once, the previous file will be overridden by default during automation. -->
```
```suggestion
       		<!-- When a user saves a Unix file more than once, the previous file will be overridden by default during automation. -->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_basic.xml:
```diff
@@ -0,0 +1,76 @@
+	The following section contains the workflow meta-data. It is a general
+	type workflow. The description is used as the basis for a default
+	workflow name when the file is imported into the workflow application.
+	The workflow name can be modified by the user during the import operation.
```
"during the import operation"
What do we mean by "import operation"?
Don't we generally refer to the action of creating a workflow as simply "creating a workflow"?

zhaocaih replied:
It was updated with the suggestion, please help review.

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfMD5.xml:
```diff
@@ -0,0 +1,73 @@
+	<skills>System Programmer</skills>
+</step>
+ 
+<!-- This step contains only instructions for user to know what should do in this step. -->
```
```suggestion
<!-- This step only contains an instructions element to inform a user what should be done for this step. -->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml:
```diff
@@ -0,0 +1,94 @@
+                          xsi:noNamespaceSchemaLocation="../workflow/schemas/workflow_v1.xsd"> 
+                          
+<!-- +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
+	 This is a sample called workflow that will be referred in workflow_sample_wf2wf.xml to demonstrate variable mapping
```
```suggestion
	 This is a sample called workflow that will be referred in workflow_sample_wf2wf.xml to demonstrate the variable mapping
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml:
```diff
@@ -0,0 +1,94 @@
+<variable name="demoString">
+	<label>Variable demoString</label>
+	<abstract>Simple variable named demoString.</abstract>
+	<description>This is a simple string variable named demoString whose length exceeds 10 characters.</description>
```
This mentions a 10 character minimum but I don't see that variable restriction enforced here

zhaocaih replied:
The description was updated, please review again, thanks.

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml:
```diff
@@ -0,0 +1,94 @@
+	</string>
+</variable>
+ 
+<!-- A step for user to input variables and tranfer its value between calling workflow and called workflow.
```
> A step for user to

 A step for _a_ user to

zhaocaih replied:
Changed as the suggestion, thanks.

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml:
```diff
@@ -0,0 +1,94 @@
+</variable>
+ 
+<!-- A step for user to input variables and tranfer its value between calling workflow and called workflow.
+		If the variable has already been set a value in the calling workflow, the value will be displayed in this step.
```
```suggestion
		If the variable value has already been set in the calling workflow, the value will be displayed in this step.
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml:
```diff
@@ -0,0 +1,94 @@
+-->
+<step name="inputVariable">
+      <title>Input Variable Step</title>
+      <description>User can input variable values in this step to transfer it to calling workflow, or get variable value from the calling workflow.</description>
```
```suggestion
      <description>The user can input variable values in this step to transfer it to calling workflow, or get variable value from the calling workflow.</description>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_calledwfVarMapping2.xml:
```diff
@@ -0,0 +1,64 @@
+-->
+<step name="inputVariable">
+      <title>Input Variable Step</title>
+      <description>User can input variable values in this step to transfer it to calling workflow, or get variable value from the calling workflow.</description>
```
```suggestion
      <description>The user can input variable values in this step to transfer it to calling workflow, or get variable value from the calling workflow.</description>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_feedback.xml:
```diff
@@ -0,0 +1,117 @@
+</feedbackItem>
+ 
+<!-- ================= step with feedback items definitions ================= -->
+ 
```
Is there a missing step  here or is this trying to show that all steps below this point are using feedback?
Seems odd since the other steps are prefaced with a comment explaining what they are demonstrating.

zhaocaih replied:
Here is no step missing. The comment “step with feedback items definitions” is corresponding to the comment “step without feedback item definitions”, this’s trying to show that the steps below this point are using feedback, and each step also have its own comment.
![feedback](https://user-images.githubusercontent.com/34873639/60486064-07889400-9cd1-11e9-81b0-50db3672dd0c.jpg)

Conner-IBM commented on zOSMF-Samples/workflow_sample_fragment0.xml:
```diff
@@ -0,0 +1,42 @@
+<step name="Deploy-certs">
+	<title>Deploy digital certificates for Function X</title>
+	<description>Create and deploy the digital certificates that are used to protect the data that Function X sends across the network.</description>
+	<instructions>In the Instructions area, describe in detail what the user needs to do to complete the step. Include any specific values the user needs to know, based on your requirements for the work to be completed. If you are directing the user to enter a command such as RACF RACDCERT, for example, use this element to state which values should be included on the command invocation. Other inform ...
```
this could use a line break or two for readability

zhaocaih replied:
line breaks were added.

Conner-IBM commented on zOSMF-Samples/workflow_sample_output.xml:
```diff
@@ -0,0 +1,110 @@
+	<title>Submit empty JCL</title>
+	<description>Submit an empy JCL job using IEFBR14.</description>
+	<variableValue name="st_user"/>
+	<instructions>This is a very simple JCL job submittion test setting1:${global-setting1} using IEFBR14 and nothing will be done here.</instructions>
```
This looks like it should have substitution set to true

zhaocaih replied:
Yes, added substitution="true".

Conner-IBM commented on zOSMF-Samples/workflow_sample_parallel_steps.xml:
```diff
@@ -0,0 +1,256 @@
+echo date 2018-02-05 >> $_output
+		</inlineTemplate>
+		<submitAs maxRc="0">shell-JCL</submitAs>
+		<!-- needResolveConflicts=false indicates how potential variable conflicts will be resolved: The workflow will use the variables from the output file. -->
```
> The workflow will use the variables from the output file. -->

May want to make this more clear that this is the behavior when needResolveConflicts is false. Also would be good to explain what happens when it is true.

Rough suggestion:
```suggestion
 <!-- needResolveConflicts indicates how potential variable conflicts will be resolved:
When equal to false, The workflow will use the variables from the output file without asking the user for input.
-->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_parallel_steps.xml:
```diff
@@ -0,0 +1,256 @@
+         In a parallel-steps workflow, the failure of an automated step does 
+         not stop automation processing for the other automated steps. 
+         Processing continues until all of the automated steps are completed 
+         or failed, or the user stops automation processing by using the 
```
>Processing continues until all of the automated steps are completed 
         or failed, or the user stops automation processing by using the 
         Stop Automation action on the Workflows page.

using "or failed, or the user" here is strange to read.
I'd suggest:
         
```suggestion
        or failed. The user can also stop automation processing by using the 
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_parallel_steps.xml:
```diff
@@ -0,0 +1,256 @@
+echo setting0 global001 >> $_output
+		</inlineTemplate>
+		<submitAs maxRc="0">shell-JCL</submitAs>
+		<!-- needResolveConflicts default is true. The "Input Variables" tab will be enabled for the user to resolve variable conflicts. -->
```
Does this stop automation? If so, it should be noted

zhaocaih replied:
Yes, the automation will stop. Update the comments as "<!-- needResolveConflicts default is true. The automation will stop and the "Input Variables" tab will be enabled for the user to resolve variable conflicts. -->"

Conner-IBM commented on zOSMF-Samples/workflow_sample_rexx_template0.txt:
```diff
@@ -0,0 +1,20 @@
+# See the License for the specific language governing permissions and       #
+# limitations under the License.                                            #
+#***************************************************************************#
+#
```
A comment explaining what sample definition this file is used with would be helpful

Conner-IBM commented on zOSMF-Samples/workflow_sample_substeps.xml:
```diff
@@ -0,0 +1,139 @@
+			<step name="subStep1_1_1">
+				<title>Third-level step</title>
+				<description>This is a third-level step.</description>
+				<!-- A step cannot pre-req itself or any ancestor step -->
```
What is meant by 'ancestor step'? I don't recall seeing this terminology used.

zhaocaih replied:
This is a concept that comes from XML which means the parent and all higher level XML elements to the current element. In workflow, 'ancestor step' means the parent step or all higher level steps of the current step.

Conner-IBM replied:
Ok, I wasn't aware of the XML definition. I would have excepted to use the term parent step, as that's how it is generally referred to in workflow - But if it has a generally understood XML meaning I'll consider this closed.

Conner-IBM commented on zOSMF-Samples/workflow_sample_variables.xml:
```diff
@@ -0,0 +1,706 @@
+	<description>This sample shows a string that is primed with the default value specified in the XML. Note that this variable is designated as required within the step. As such, the variable is displayed somewhat differently than the others.	
+	</description>
+	<category>String Variables</category>
+	<!-- The text box is primed with the default value. -->
```
```suggestion
	<!-- The text box in the Workflows task primed with the default value if no other value is provided. -->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_variables.xml:
```diff
@@ -0,0 +1,706 @@
+</variable>
+ 
+<!-- 
+	 Here, we define steps which reference variables, thus causing the Workflows task
```
A variable does not need to be defined in a step for a step to use the variable for substitution.
People misunderstanding this leads to them believing they need to define every variable a step uses, in every step; which is tedious and unnecessary work.

zhaocaih replied:
We added a step without variable defined in it and also added comment to clarify the point, please help review again, thanks.

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+		As the element calledWorkflowDefinitionFile is NOT specified in this step, user has to input their own valid definition file path when prompt for user to create the called workflow instance.
+		The element calledWorkflowID is specified to indicate that the called workflow instance should has the same workflowID in its definition file.
+		Here the called workflow will be workflow_sample_calledwfBasic.xml.
+		If a instance with the specified calledWorkflowID already exist, workflow will directly open this instance when user start perform it, if the instance does not exist or multiple instances
```
```suggestion
		If a instance with the specified calledWorkflowID already exists, workflow will directly open this instance when user start perform it, if the instance does not exist or multiple instances
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+	 -->
+<step name="simpleWf2wf">
+	<title>A very simple sample of workflow to workflow execution.</title>
+	<description>This is a very simple demo of using workflow to worfklow execution in which only the one mondatory element named calledWorkflowID is specified.</description>
```
```suggestion
	<description>This is a very simple demo of using workflow to workflow execution in which only the one mandatory element named calledWorkflowID is specified.</description>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+	<calledWorkflowID>workflow.sample.calledwfBasic</calledWorkflowID>
+</step>
+ 
+<!-- This step is used for user to input below variable values some of which will be used to transfer to the following called workflow:
```
```suggestion
<!-- This step is used for the user to input the following variable values. Some of which will transfered to called workflow:
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+ -->
+<step name="prepareVariables" >
+	<title>Prepare Variable Step</title>
+	<description>In this step, you enter some variable values to see if they can be transferred to the called workflow in the following step.</description>
```
```suggestion
	<description>In this step, you enter variable values, some of which will be transferred to the called workflow in the following step.</description>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+	<variableValue name="String1"/>
+	<variableValue name="String2"/>
+	<variableValue name="sampleString"/>
+	<instructions>User can input some values for the variables in this step, those values can be transfered to the called workflow if they satisfy the mapping rules that user specified in the following step.</instructions>
```
```suggestion
	<instructions>The user can input some values for the variables in this step, those values can be transferred to the called workflow if they satisfy the mapping rules that user specified in the following step.</instructions>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+		Additionally, <variableName> element will map the variable "sampleString" in this calling workflow to the variable "demoString" in the called workflow
+		and transfer its value to demoString in called workflow.
+		
+		Note: When user specify the XML definition and create the called workflow successfully, the variables that meet the mapping rules will be transferred
```
```suggestion
		Note: When a user specifies the XML definition and creates the called workflow successfully, the variables that meet the mapping rules will be transferred
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+		
+		Note: When user specify the XML definition and create the called workflow successfully, the variables that meet the mapping rules will be transferred
+		from the calling workflow to the called workflow only once, that means, no matter how many times the user perform the calling step in future, those 
+		variable values will not be transferred any more. But if user decide to change the existing called workflow, this will be treated as creating a new called
```
```suggestion
		variable values will not be updated. But if user decide to change the existing called workflow, this will be treated as creating a new called
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+</step>
+ 
+<!-- This step demonstrates the use of variableMapping element for user to transfer variable values from called workflow to calling workflow.
+		The element <variableName> under the element <fromCalledToCalling indicates that when the called workflow become Complete status, 
```
```suggestion
		The element <variableName> under the element <fromCalledToCalling> indicates that when the called workflow changes to the "Complete" state, 
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+ 
+<!-- This step demonstrates the use of variableMapping element for user to transfer variable values from called workflow to calling workflow.
+		The element <variableName> under the element <fromCalledToCalling indicates that when the called workflow become Complete status, 
+		the variable "userID" in the called workflow will be mapped to the variable "st_user" in the calling workflow, additionally, the variable
```
```suggestion
		the variable "userID" in the called workflow will be mapped to the variable "st_user" in the calling workflow. Additionally, the variable
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+<!-- This step demonstrates the use of variableMapping element for user to transfer variable values from called workflow to calling workflow.
+		The element <variableName> under the element <fromCalledToCalling indicates that when the called workflow become Complete status, 
+		the variable "userID" in the called workflow will be mapped to the variable "st_user" in the calling workflow, additionally, the variable
+		value will also be set to st_user in the calling workflow when the called workflow become to Complete status.
```
```suggestion
		value will also be set to st_user in the calling workflow when the called workflow changes to the "Complete" state. 
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+		value will also be set to st_user in the calling workflow when the called workflow become to Complete status.
+		
+		Note: When the called workflow is performed and become Complete status, the variables that meet the mapping rules will be transferred
+		from the called workflow to the calling workflow, this will also happen only once no matter how the status of the called workflow change
```
```suggestion
		from the called workflow to the calling workflow, this will also happen only once - even if the state of the called workflow changes later on.
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+		the variable "userID" in the called workflow will be mapped to the variable "st_user" in the calling workflow, additionally, the variable
+		value will also be set to st_user in the calling workflow when the called workflow become to Complete status.
+		
+		Note: When the called workflow is performed and become Complete status, the variables that meet the mapping rules will be transferred
```
```suggestion
		Note: When the called workflow is performed and changes to the "Complete" state, the variables that meet the mapping rules will be transferred
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+	<calledWorkflowID>workflow.sample.calledwfVarMapping2</calledWorkflowID>
+	<calledWorkflowVersion>1.0</calledWorkflowVersion>
+</step>
+<!-- User can check if the value of variable st_user is already transferred from the called workflow in the above step varMappingFromCalled2Calling -->
```
```suggestion
<!-- User can check if the value of variable st_user has already been transferred from the called workflow in the above step varMappingFromCalled2Calling -->
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+	<title>Check Variable from Called Workflow</title>
+	<description>This is a simple step to display the value of variable st_user in the current calling workflow.</description>
+	<variableValue name="st_user"/>
+	<instructions>In this step, user can check if the value of variable st_user is successfully transferred from the called workflow in previous step.</instructions>
```
```suggestion
	<instructions>In this step, the user can check if the value of variable st_user has been successfully transferred from the called workflow in previous step.</instructions>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wf2wf.xml:
```diff
@@ -0,0 +1,203 @@
+-->
+<step name="demoMD5">
+	<title>Demo of Using MD5.</title>
+	<description>In this step, a MD5 code is used to indicate the appropriate called workflow.</description>
```
```suggestion
	<description>In this step, a MD5 hash is used to indicate the appropriate called workflow.</description>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards.xml:
```diff
@@ -0,0 +1,259 @@
+	<abstract>High-level qualifier.</abstract>
+	<description>The data set high-level qualifier (HLQ) to be used for saving the generated JCL.</description>
+	<category/>
+	<!-- The default value is "General". --> 
```
```suggestion
	<!-- The default value for the category element is "General". --> 
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards.xml:
```diff
@@ -0,0 +1,259 @@
+</variable>
+ 
+<!-- This is a manual step that demonstrates the use of variable substitution within the 
+     manual instructions. 
```
```suggestion
     instructions. 
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards.xml:
```diff
@@ -0,0 +1,259 @@
+        <variableValue name="hlq"/>
+		<instructions substitution="false">In this example step, you enter a RACF command through a JCL job.
+		<br/>
+		You will have the option to save a copy of the JCL <em>after</em> you have reviewed it for your approval.</instructions>
```
```suggestion
		You will have the option to save a copy of the JCL <em>after</em> you have reviewed it.</instructions>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards_upgrade.xml:
```diff
@@ -0,0 +1,327 @@
+		   In the workflowNotes section, nothing is provided, so not checked it
+		-->
+		
+		<upgradeNotes>Upgrade from version 1.0 to 2.0 and not cancel version 1.0</upgradeNotes>
```
```suggestion
		<upgradeNotes>Upgrade from version 1.0 to 2.0 and do not cancel version 1.0</upgradeNotes>
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards_upgrade.xml:
```diff
@@ -0,0 +1,327 @@
+		 -->	
+		<workflowHistory defaultChecked = "true"/>
+		<!--  
+		   In the workflowNotes section, nothing is provided, so not checked it
```
What is this comment trying to say? It looks like defaultChecked is true, but it says "so not checked it"?

zhaocaih replied:
It's talking about the workflowNotes section, but no such an element is defined in this sample, so removed this comment.

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards_upgrade.xml:
```diff
@@ -0,0 +1,327 @@
+		-->
+		<!--
+			In the workflowNotes section, if the defaultChecked is true, the UI will display the 
+			workflow notes check box which checked status.
```
```suggestion
			workflow notes check box checked.
```

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards_upgrade.xml:
```diff
@@ -0,0 +1,327 @@
+		
+		<upgradeNotes>Upgrade from version 1.0 to 2.0 and not cancel version 1.0</upgradeNotes>
+	</version>
+	<!-- The type is set to "replace" in this version section to indicate  cancelling the old version 2.0 instance after upgrade successfully. -->
```
Im also confused by this comment. Isn't this version 2.0? What does it mean by canceling old version 2.0?

zhaocaih replied:
This comment is to explain the following section: <version value="2.0" type="replace”>, it indicates the based workflow instance version is 2.0. 
During the upgrade, it will create a new instance and copy variable values from the based instance, and cancel the original 2.0 workflow instance.

Conner-IBM commented on zOSMF-Samples/workflow_sample_wizards_upgrade.xml:
```diff
@@ -0,0 +1,327 @@
+	 definition, choose whether cancel the old workflow or not after the upgrade complete
+	 by specifying the attribute "type" in their "version" element.
+	 
+	 Note, this sample of upgrade workflow is supposed to upgrade the workflow that created
```
Since this is a copy of the workflow_sample_wizards, same my suggestions apply to this file as the other one.

zhaocaih replied:
The same changed made for workflow_sample_wizards also made on workflow_sample_wizards_upgrade.

Conner-IBM reviewed (changes requested):
Mainly grammatical concerns, but a few issues with the content as well.
I'd like to see these addressed before publishing these files as part of the project.

shreeji818 reviewed (changes requested):
I had discussion with John Mertic from OMP team. He observed that we have IBM copyright statement and suggested that we should change entire copy right block as below. This provides simpler copyright than what I communicated earlier.

/************************************************************************
/* Copyright Contributors to the zOS-Workflow Project.
/*
/*  SPDX-License-Identifier: Apache-2.0
/************************************************************************

zhaocaih commented:
Hi Hiren, Conner, Thanks for the comments, very helpful. It's inconvenient to track so many comments in one single request, I'd like to submit pull request separately after addressed the comments for each sample. what do you think?

Conner-IBM commented:
@zhaocaih , I'm not sure if I'm understanding correctly. Are you asking for this pull request to be accepted as is, and then at some later point fix the comments in this review in individual pull requests per sample?
If so, I'd rather keep all the comments and their respective fixes all together if possible rather than accepting a pull request into the repo with open issues.

I'm not terribly familiar with Git processes yet; But can't this pull request be updated with as many additional commits as required in order to address the comments? Each of these commits could in address a single file if that is more convenient. For some of the simple grammatical comments it seems like you can even just accept suggestions through the webtool.

zhaocaih commented:
@Conner-IBM Sorry I didn't make myself clear in my previous comments. I meant close (reject) this pull request without merge. When I addressed the comments for one file, I will submit a new pull request for that single file, I think it could be more clearly and easier to track and decide if accept for that single file, and we can make progress step by step.
This pull request now include 28 files, it may take longer time to address all comments, as some may need a test or discussion before make changes (I include them in one PR because I thought only copyright need update at first).
 
Each of these commits could in address a single file sounds a good idea, I am just afraid that as the conversation number goes up it gets harder to read. 
I am not very familiar with Git, I will try from accepting grammatical suggestions through the webtool to see how it will go.

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_automation.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_automation.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_calledwfMD5.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_calledwfVarMapping1.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_calledwfVarMapping2.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_parallel_steps.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_parallel_steps.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_variables.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wf2wf.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wizards.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wizards.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wizards.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wizards_upgrade.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_wizards_upgrade.xml

Co-Authored-By: Conner-IBM <50423738+Conner-IBM@users.noreply.github.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update copyright

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih commented:
@shreeji818 I have updated the copyright.

zhaocaih committed:
update workflow_sample_basic.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_calledwfVarMapping1.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
add line breaks

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_parallel_steps.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_output.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_wizards_upgrade.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_rexx_template0.txt

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_file_template0.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_wf2wf.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_output.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_variables.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

Conner-IBM commented on zOSMF-Samples/workflow_sample_output.xml:
```diff
@@ -0,0 +1,101 @@
+</variable>
+
+<!-- This step is a demo to copy a z/OS UNIX file into an MVS data set. 
+In this sample, dataset "ZOSMF.OUTPUT" is used, the user should create this dataset first or use an existing one.-->  
```
```suggestion
In this sample, dataset "ZOSMF.OUTPUT" is used in the output element. The user should create this dataset first or change the value of the output element to an existing data set.-->  
```

Alternatively, it may be easier to use a unix file in /tmp/ for output to avoid having to use datasets.

zhaocaih committed:
Update zOSMF-Samples/workflow_sample_output.xml

Co-Authored-By: Conner-IBM <conner.ferguson@ibm.com>
Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih committed:
Update workflow_sample_variables.xml

Signed-off-by: Caihua Zhao <zhaocaih@cn.ibm.com>

zhaocaih commented:
@shreeji818 Hiren, all the comments of this pull request were addressed, the copyright was updated, can you please review?

shreeji818 commented:
I have approved the changes. Li Qi should be able to merge the changes into the main branch after resolving the conflict.

shreeji818 commented:
We need to update the Readme.md file in this folder to describe each sample. Will you please take care of that before merging? Otherwise you will need a new pull request to update the ReadMe file.

ibmliqi committed:
Merge branch 'master' into zch-uploadSamples
